src/aggregate/aggregate_exec.c

```diff
@@ -974,7 +975,8 @@ static int buildPipelineAndExecute(AREQ *r, RedisModuleCtx *ctx, QueryError *sta
       rs_timersub(&time, &r->qiter.initTime, &time);
       rs_timeradd(&time, &r->qiter.GILTime, &r->qiter.GILTime);
     }
-    workersThreadPool_AddWork((redisearch_thpool_proc)AREQ_Execute_Callback, BCRctx);
+    const int rc = workersThreadPool_AddWork((redisearch_thpool_proc)AREQ_Execute_Callback, BCRctx);
+    RS_ASSERT_ALWAYS(rc == 0);
```

Comment:
Why do we want to assert that (and always)?

Reply:
`workersThreadPool_AddWork` only returns non-zero on memory allocation failure

Reply:
Since it cannot happen today, I think it is better to have `RS_ASSERT` only, until we have a reason to use it always